### ISSUE TITLE
Use IAM Role rather then AK/SK for accessing AWS Bedrock

### ISSUE THREAD

**Issue by yubingjiaocn** [langgenius/dify#3471]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I'm trying to use IAM Role associated with EC2 instance to authenticate against AWS Bedrock. When I'm trying to add AWS Bedrock model provider, I found `Access Key` and `Secret Key` field is mandatory. 

### 2. Describe the feature you'd like to see

Add an option for Dify frontend to ignore AK/SK and let AWS SDK use IAM role associated with instance (or container if running on Amazon ECS/EKS) directly. 

### 3. How will this feature improve your workflow or experience?

Enhance security by adhering security best practice of not using long-lived credential. 

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by biosugar0** [comment]
Hi team,

When initializing a client with Boto3 and AnthropicBedrock without specifying `aws_access_key_id` and `aws_secret_access_key`, the default AWS credential provider will be used. Here's an example:

```python
from anthropic import AnthropicBedrock

client = AnthropicBedrock(
    # Do not specify keys
    aws_region="us-west-2",
)
```

In this case, the SDK will search for IAM Role credentials in the following order:

1. Environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
2. `~/.aws/credentials` file
3. IAM Role attached to the EC2 instance or ECS task

Therefore, if you assign an IAM Role to your application, the SDK will automatically obtain the temporary credentials of that role and sign the API requests. This eliminates the need to embed keys in the code, enhancing security by adhering to best practices.

In conclusion, by making `aws_access_key_id` and `aws_secret_access_key` optional and allowing the SDK to select the credentials, this issue can be resolved.
